### ISSUE TITLE
missing dropout on the embeddings

### ISSUE THREAD

**Issue by JulesGM** [jadore801120/attention-is-all-you-need-pytorch#69]
There should be dropout on the embeddings, before feeding them to the encoder layers. See the reference implementation:
https://github.com/tensorflow/models/blob/master/official/transformer/model/transformer.py#L124
and see its absence here:
https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/master/transformer/Models.py#L87

Otherwise your code is really awesome, keep it up.

**Comment by huangnengCSU** [comment]
I also think there should be a dropout layer between the positionalEncoding and first encoder module.

**Comment by jadore801120** [comment]
@JulesGM 
Thanks for your pull request and pointing out! Sincerely appreciate that!
However, I made a huge modification before check the pull requests, so it is hard to merge now. Sorry about that. 😓
Best, 
Yu-Hsiang
